django/thunderstore/api/cyberstorm/views/team.py

```diff
@@ -0,0 +1,46 @@
+from django.http import Http404, HttpRequest, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.api.cyberstorm.serializers import (
+    CyberstormTeamMemberSerializer,
+    CyberstormTeamSerializer,
+)
+from thunderstore.repository.models.team import Team
+
+
+class TeamAPIView(APIView):
+    permission_classes = []
+
+    @swagger_auto_schema(
+        responses={200: CyberstormTeamSerializer()},
+        operation_id="api.team",
+    )
+    def get(self, request: HttpRequest, team_identifier: str) -> HttpResponse:
+        try:
+            t = Team.objects.prefetch_related("members",).get(
+                name=team_identifier,
+            )
+
+        except Team.DoesNotExist:
+            raise Http404(f"Coudln't find team with the identifier {team_identifier}")
+
+        serializer = self.serialize_results(t)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def serialize_results(self, team: Team):
+        return CyberstormTeamSerializer(
+            {
+                "name": team.name,
+                "members": [
+                    CyberstormTeamMemberSerializer(
+                        {"user": team_member.user.username, "role": team_member.role}
```

Comment:
Usernames are bad user identifiers since they can be changed. See comment about expanding the serializer to encode users as objects with an ID and username.